scripts/verify.sh

```
@@ -0,0 +1,79 @@
+# Local mirror of the CI gate (.github/workflows/build-and-test.yml).
+# Closes the "build passed locally but CI failed" gap: runs the SAME commands CI
+# runs, so charset/format, integration, casing, and drift surface here, not in CI.
+#
+# Usage: scripts/verify.sh [fast|full]   (default: fast)
+#   fast — build + dotnet format --verify + frontend ci+test + doc drift   (no Docker)
+#   full — fast scope + full `dotnet test` (Testcontainers; needs Docker), exactly like CI
+#
+# Only the layers whose files changed (vs origin/main) run, so doc-only work stays quick.
+# On Windows run via Git Bash. Wired as the pre-push hook (scripts/hooks/pre-push).
+set -uo pipefail
+cd "$(dirname "$0")/.."
```

Shellcheck (0.11.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)